etstool.py

```diff
@@ -0,0 +1,288 @@
+#  under the conditions described in the aforementioned license.  The license
+#  is also available online at http://www.enthought.com/licenses/BSD.txt
+#
+#  Thanks for using Enthought open source!
+#
+"""
+Tasks for Test Runs
+===================
+
+This file is intended to be used with a python environment with the
+click library to automate the process of setting up test environments
+and running the test within them.  This improves repeatability and
+reliability of tests be removing many of the variables around the
+developer's particular Python environment.  Test environment setup and
+package management is performed using `EDM
+http://docs.enthought.com/edm/`_
+
+To use this to run you tests, you will need to install EDM and click
+into your working environment.  You will also need to have git
+installed to access required source code from github repositories.
+You can then do::
+
+    python etstool.py install --runtime=...
+
+to create a test environment from the current codebase and::
+
+    python etstool.py test --runtime=...
+
+to run tests in that environment.  You can remove the environment with::
+
+    python etstool.py cleanup --runtime=...
+
+If you make changes you will either need to remove and re-install the
+environment or manually update the environment using ``edm``, as
+the install performs a ``python setup.py install`` rather than a ``develop``,
+so changes in your code will not be automatically mirrored in the test
+environment.  You can update with a command like::
+
+    edm run --environment ... -- python setup.py install
+
+You can run all three tasks at once with::
+
+    python etstool.py test_clean --runtime=...
+
+which will create, install, run tests, and then clean-up the environment.  And
+you can run tests in all supported runtimes::
+
+    python etstool.py test_all
+
+Currently supported runtime values are ``2.7`` and ``3.5``.  Not all
+combinations of runtimes will work, but the tasks will fail with
+a clear error if that is the case.
+
+Tests can still be run via the usual means in other environments if that suits
+a developer's purpose.
+
+Changing This File
+------------------
+
+To change the packages installed during a test run, change the dependencies
+variable below.  To install a package from github, or one which is not yet
+available via EDM, add it to the `ci-src-requirements.txt` file (these will be
+installed by `pip`).
+
+Other changes to commands should be a straightforward change to the listed
+commands for each task. See the EDM documentation for more information about
+how to run commands within an EDM enviornment.
+
+"""
+
+import glob
+import os
+import platform
+import subprocess
+import sys
+from shutil import rmtree, copy as copyfile
+from tempfile import mkdtemp
+from contextlib import contextmanager
+
+import click
+
+dependencies = {
+    "coverage",
+    "cython",
+    "Sphinx",
+    "mock",
+    "nose",
+    "numpy",
+    "traitsui"
```

Comment:
For CI purposes, I think we need a Qt backend too (for `traits.tests.test_ui_notifiers.TestDecoratorUINotifiers`). PyQt?